org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/TestProcess.xtend

```diff
@@ -0,0 +1,62 @@
+package org.testeditor.web.backend.testexecution
+
+import static org.testeditor.web.backend.testexecution.TestStatus.*
+
+class TestProcess {
+
+	public static val DEFAULT_IDLE_TEST_PROCESS = new TestProcess()
+	public static val WAIT_TIMEOUT_SECONDS = 5
+
+	var Process process
+	var TestStatus status
+
+	new(Process process) {
+		if (process === null) {
+			throw new NullPointerException("Process must initially not be null")
+		}
+		this.process = process
+		this.status = RUNNING
+	}
+
+	private new() {
+		this.process = null
+		this.status = IDLE
+	}
+
+	def TestStatus getStatus() {
+		val processRef = this.process
+		if (processRef !== null && !processRef.alive) {
+			markCompleted(processRef.exitValue)
+		}
+		return status
+	}
+
+	def TestStatus waitForStatus() {
+		val processRef = this.process
+		if (processRef !== null && processRef.alive) {
+			processRef.waitFor.markCompleted
+		}
+		return this.getStatus
+	}
+
+	private static def TestStatus getStatus(int exitCode) {
```

Comment:
how about 'toTestStatus', since it is more of a conversion (and it won't carry the same name as the api 'getStatus' method).
is there some need to have it static?

I'd put this method at the end of the class, since it is used in markCompleted only, right?

Reply:
The only reason why it was static is that there was no reason for it to be an instance method (it operates only on its parameters)…